### DISCUSSION TITLE
Showing the short url as origin of the traffic in stats of the long url's site

### DISCUSSION THREAD

**Discussion by ryanb3n** [YOURLS/YOURLS#3783] [General]
Is there any way to enable referral url to show on site's stats the short url linked for?
For example, when visiting the a site using a short url shared on social media, I can see the traffic came from social media. However, using the short url in the browser, the site stats don't show the origin of the traffic. I would like it to show that it came from the specific short url.

**Comment by dgw** [comment]
Directly-entered URLs are not supposed to carry "referrer" data. You can read more about HTTP referrer semantics [in RFC 9110](https://www.rfc-editor.org/rfc/rfc9110#field.referer):

> The "Referer" [sic] header field allows the user agent to specify a URI reference for the resource from which the [target URI](https://www.rfc-editor.org/rfc/rfc9110#target.resource) was obtained (i.e., the "referrer", though the field name is misspelled).
>
> […]
>
> If the target URI was obtained from a source that does not have its own URI (e.g., input from the user keyboard, or an entry within the user's bookmarks/favorites), the user agent **MUST** either exclude the Referer header field or send it with a value of "about:blank".

Therefore, if a user enters the shorturl manually, that request is considered to have no referrer.

You will need to add a parameter(s) to the long URLs stored in YOURLS, which your analytics solution can use as the source (if no referrer is sent); the HTTP referrer isn't an option here.

Adding these tags can be automated with a plugin; for example, one longstanding entry on [our Awesome List](https://github.com/YOURLS/awesome) (though not personally tested) is [this Google Analytics Link Tagging plugin](https://katz.co/2300/) ([GH repo](https://github.com/katzwebservices/google-analytics-for-yourls)). If that plugin is not sufficiently configurable for your use case, you can likely use it as a starting point for your own link auto-tagger.

**Comment by ryanb3n** [comment]
Understood. Thank you for the detailed explanation!
